apps/studio/src/pages/sites/[siteId]/folders/[folderId]/index.tsx

```diff
@@ -109,23 +143,205 @@ const FolderPage: NextPageWithLayout = () => {
         siteId={siteId}
-        key={uniqueId()}
+        resourceId={parseInt(folderId)}
       />
     </>
   )
 }
 
 FolderPage.getLayout = AdminCmsSidebarLayout
 export default FolderPage
+
+interface FolderSettingsModalProps {
+  isOpen: boolean
+  onClose: () => void
+  siteId: string
+  resourceId: number
+}
+const FolderSettingsModal = ({
+  isOpen,
+  onClose,
+  siteId,
+  resourceId,
+}: FolderSettingsModalProps) => {
+  return (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <ModalOverlay />
+      <SuspendableModalContent
+        isOpen={isOpen}
+        siteId={siteId}
+        resourceId={resourceId}
+        onClose={onClose}
+      />
+    </Modal>
+  )
+}
+
+const SuspendableModalContent = ({
+  isOpen,
+  onClose,
+  siteId,
+  resourceId,
+}: FolderSettingsModalProps) => {
+  const { setValue, register, handleSubmit, watch, formState, getFieldState } =
+    useZodForm({
+      defaultValues: {
+        title: "",
+        permalink: "",
+      },
+      schema: editFolderSchema.omit({ siteId: true, resourceId: true }),
+    })
+  const [{ title: originalTitle, permalink: originalPermalink, parentId }] =
+    trpc.folder.readFolder.useSuspenseQuery({
+      siteId: parseInt(siteId),
+      resourceId,
+    })
+  const { errors, isValid } = formState
+  const utils = trpc.useUtils()
+  const toast = useToast()
+  const { mutate, isLoading } = trpc.folder.editFolder.useMutation({
+    onSettled: onClose,
+    onSuccess: () => {
+      void utils.site.list.invalidate()
+      void utils.resource.list.invalidate()
+      void utils.resource.getChildrenOf.invalidate({
+        resourceId: parentId ? String(parentId) : null,
+      })
+      void utils.folder.readFolder.invalidate()
+      toast({ title: "Folder created!", status: "success" })
+    },
+    onError: (err) => {
+      toast({
+        title: "Failed to create folder",
+        status: "error",
+        // TODO: check if this property is correct
+        description: err.message,
+      })
+    },
+  })
+
+  const onSubmit = handleSubmit((data) => {
+    mutate({ ...data, resourceId: String(resourceId), siteId })
+  })
+
+  const [title, permalink] = watch(["title", "permalink"])
+
+  useEffect(() => {
+    const permalinkFieldState = getFieldState("permalink")
+    // This allows the syncing to happen only when the page title is not dirty
+    // Dirty means user has changed the value AND the value is not the same as the default value of "".
+    // Once the value has been cleared, dirty state will reset.
+    if (!permalinkFieldState.isDirty) {
+      setValue("permalink", generateResourceUrl(title || ""), {
+        shouldValidate: !!title,
+      })
+    }
+  }, [getFieldState, setValue, title])
+
+  useEffect(() => {
+    if (originalPermalink) {
+      setValue("title", originalTitle)
+    }
+
+    if (originalPermalink) {
+      setValue("permalink", originalPermalink)
+    }
+  }, [originalPermalink, originalTitle])
```

Comment:
if we set the defaults immediately when we fetch the folder metadata, there is no need for this effect anymore

Reply:
done